src/app/component/dashboard/customers/customers.component.html

```diff
@@ -43,33 +46,85 @@
+      [hidden]="hiddenColumns.indexOf('shortname') > -1">
+      <ng-template kendoGridHeaderTemplate let-dataItem>
+        {{dataItem.field}}
+        <button (click)="onEvent($event); hideColumn(dataItem.field)" class="k-button k-primary hide-b">
+          X
+        </button>
+      </ng-template>
     </kendo-grid-column>
-    <kendo-grid-column field="firstname" title="{{ language.firstname }}" width="120">
+    <kendo-grid-column field="firstname" title="{{ language.firstname }}" width="120"
+      [hidden]="hiddenColumns.indexOf('firstname') > -1">
+      <ng-template kendoGridHeaderTemplate let-dataItem>
+        {{dataItem.field}}
+        <button (click)="onEvent($event); hideColumn(dataItem.field)" class="k-button k-primary hide-b">
+          X
+        </button>
+      </ng-template>
     </kendo-grid-column>
-    <kendo-grid-column field="lastname" title="{{ language.lastname }}" width="120">
+    <kendo-grid-column field="lastname" title="{{ language.lastname }}" width="120"
+      [hidden]="hiddenColumns.indexOf('lastname') > -1">
+      <ng-template kendoGridHeaderTemplate let-dataItem>
+        {{dataItem.field}}
+        <button (click)="onEvent($event); hideColumn(dataItem.field)" class="k-button k-primary hide-b">
+          X
+        </button>
+      </ng-template>
     </kendo-grid-column>
-    <kendo-grid-column field="telephone" title="{{ language.telephone }}" width="120">
+    <kendo-grid-column field="telephone" title="{{ language.telephone }}" width="120"
+      [hidden]="hiddenColumns.indexOf('telephone') > -1">
+      <ng-template kendoGridHeaderTemplate let-dataItem>
+        {{dataItem.field}}
+        <button (click)="onEvent($event); hideColumn(dataItem.field)" class="k-button k-primary hide-b">
+          X
+        </button>
+      </ng-template>
     </kendo-grid-column>
-    <kendo-grid-column field="mobile" title="{{ language.mobile }}" width="120">
+    <kendo-grid-column field="mobile" title="{{ language.mobile }}" width="120"
+      [hidden]="hiddenColumns.indexOf('mobile') > -1">
+      <ng-template kendoGridHeaderTemplate let-dataItem>
+        {{dataItem.field}}
+        <button (click)="onEvent($event); hideColumn(dataItem.field)" class="k-button k-primary hide-b">
+          X
+        </button>
+      </ng-template>
     </kendo-grid-column>
-    <kendo-grid-column field="email" title="{{ language.email }}" width="150">
+    <kendo-grid-column field="email" title="{{ language.email }}" width="150"
+      [hidden]="hiddenColumns.indexOf('email') > -1">
+      <ng-template kendoGridHeaderTemplate let-dataItem>
+        {{dataItem.field}}
+        <button (click)="onEvent($event); hideColumn(dataItem.field)" class="k-button k-primary hide-b">
+          X
+        </button>
+      </ng-template>
     </kendo-grid-column>
-    <kendo-grid-column title="{{ language.preview }}" width="100">
+    <kendo-grid-column field="preview" title="{{ language.preview }}" width="110"
```

Comment:
Ako se ovo odnosi na skrivanje preview-a onda to nije dobro.
Ne smemo da sakrivamo kolone koje u sebi sadrze akcije.

Reply:
Da li to znaci da kolona koja pod sobom sadrzi neki button koji poziva neku akciju ne sme biti ponudjena za skrivanje u listi ?

Reply:
@Vladimir0610 Da. Znaci kolone koje sadrze akciju se ne skrivaju.

Reply:
Sto se tice kendo-grida sredjeno je, ali je ostao ejs-grid gde iz nekog razloga ne mogu da skinem iz Choosera buttone koje sadrze akciju. Ne znam da li ga nesto gazi ili ne, pa bih voleo taj deo da prodjemo samo.